### PR TITLE
Add CircleCI

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes https://github.com/mozilla/notes/issues/17

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+machine:
+  node:
+    version: 6
+
+test:
+  override:
+    - echo "Tests WIP"
```
Comment (pdehaan) [inline circle.yml]: Add `- npm run lint`?

**Comment by pdehaan** [comment]
Not sure if this is extreme overkill, but over in [**SnoozeTabs** land](https://github.com/bwinton/SnoozeTabs/blob/master/circle.yml), I believe we do a build of the add-on in Circle and then copy the generated XPI into the `$CIRCLE_ARTIFACTS` directory. That way we can grab old add-ons on a per-run basis by grabbing them from Circle (see https://circleci.com/gh/bwinton/SnoozeTabs/560#artifacts/containers/0).

Again, not sure if that is needed, considering how trivial it'd be to do a git checkout of a previous commit and then build it locally, but this may be useful for contributors or whatever.

**Comment by vladikoff** [comment]
@pdehaan that would be nice!

File: circle.yml (hunk 1 above)
Comment (vladikoff) [inline circle.yml]: @pdehaan that exploded with errors, so we gonna have to add that later

**Pull request merged**